airflow/callbacks/callback_requests.py

```diff
@@ -71,6 +72,7 @@ class TaskCallbackRequest(CallbackRequest):
     :param is_failure_callback: Flag to determine whether it is a Failure Callback or Success Callback
     :param msg: Additional Message that can be used for logging to determine failure/zombie
     :param processor_subdir: Directory used by Dag Processor when parsed the dag.
+    :param task_callback_type: e.g. whether on success, on failure, on retry.
```

Comment:
Is this fully covering `is_failure_callback` param? Maybe worth to remove `is_failure_callback` in that case

Reply:
Yes I have removed `is_failure_callback` and made it class property now.